File: frontend/src/types/agent.ts
export interface AgentRegisterRequest {
  name: string;
  description?: string;
  token?: string;
  app_token?: string;
}

export interface AgentRegisterResponse {
  agent_id: string;
  token: string;
  name: string;
  registered_at: string;
}

export interface AgentInfo {
  agent_id: string;
  name: string;
  description?: string;
  registered_at?: string;
  last_seen_at?: string;
  app_id?: string;
  app_name?: string;
}

File: app/models/agent.py
from __future__ import annotations
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class AgentRegisterRequest(BaseModel):
    name: str = Field(..., description="Unique agent name (e.g. otel-collector-prod)")
    source_type: str = Field(
        ...,
        description="Type of agent: otel-collector | k8s-agent | aws-agent | mock | custom",
    )
    description: Optional[str] = None


class AgentRegisterResponse(BaseModel):
    agent_id: str
    token: str
    name: str
    source_type: str
    registered_at: datetime


class AgentInfo(BaseModel):
    agent_id: str
    name: str
    source_type: str
    description: Optional[str] = None
    registered_at: Optional[datetime] = None
    last_seen_at: Optional[datetime] = None
